crates/swc_ecma_minifier/src/compress/optimize/util.rs

```diff
@@ -567,3 +600,86 @@ impl Drop for SynthesizedStmts {
         }
     }
 }
+
+#[derive(Default)]
+struct LabelAnalyzer {
+    label: JsWord,
+    top_can_break: bool,
+    count: usize,
+    break_layer: usize,
+    continue_layer: usize,
+}
+
+impl LabelAnalyzer {
+    fn visit_mut_loop(&mut self, n: &mut impl VisitMutWith<LabelAnalyzer>) {
+        self.break_layer += 1;
+        self.continue_layer += 1;
+
+        n.visit_mut_children_with(self);
+
+        self.break_layer -= 1;
+        self.continue_layer -= 1;
+    }
+}
+
+impl VisitMut for LabelAnalyzer {
+    fn visit_mut_function(&mut self, _: &mut Function) {}
+
+    fn visit_mut_class(&mut self, _: &mut Class) {}
+
+    fn visit_mut_arrow_expr(&mut self, _: &mut ArrowExpr) {}
+
+    fn visit_mut_object_lit(&mut self, _: &mut ObjectLit) {}
+
+    fn visit_mut_for_stmt(&mut self, n: &mut ForStmt) {
+        self.visit_mut_loop(n)
+    }
+
+    fn visit_mut_for_in_stmt(&mut self, n: &mut ForInStmt) {
+        self.visit_mut_loop(n)
+    }
+
+    fn visit_mut_for_of_stmt(&mut self, n: &mut ForOfStmt) {
+        self.visit_mut_loop(n)
+    }
+
+    fn visit_mut_while_stmt(&mut self, n: &mut WhileStmt) {
+        self.visit_mut_loop(n)
+    }
+
+    fn visit_mut_do_while_stmt(&mut self, n: &mut DoWhileStmt) {
+        self.visit_mut_loop(n)
+    }
+
+    fn visit_mut_switch_stmt(&mut self, n: &mut SwitchStmt) {
+        self.break_layer += 1;
+
+        n.visit_mut_children_with(self);
+
+        self.break_layer -= 1;
+    }
+
+    fn visit_mut_break_stmt(&mut self, n: &mut BreakStmt) {
+        if let Some(lb) = &n.label {
+            if lb.sym == self.label {
+                if self.break_layer > 0 || !self.top_can_break {
```

Comment:
What is the use of `top_can_break`? I can't think of a situation where it will be necessary.

Reply:
If top is a noraml break, we must preserve that labelled break because we don't have another control flow statement in JS; if it is, that label can be simply removed. So
```js
a: for (let i = 0; i < 10; i++) {
  if (i % 5 === 0) break a
  console.log(i)
}
```
can be turned into
```js
for (let i = 0; i < 10; i++) {
  if (i % 5 === 0) break
  console.log(i)
}
```

Reply:
While replying to this, I realized that you can add labels to non-loop/switch:

```js
FOO: {
  break FOO;
}
```

Here, removing `FOO` leaves `{ break }`, which is invalid.

Can we add a comment?